code/modules/antagonists/traitor/objectives/demoralise_graffiti.dm

```diff
@@ -0,0 +1,335 @@
+	icon_state = "deathcan"
+	worn_icon_state = "spraycan"
+	inhand_icon_state = "spraycan"
+	lefthand_file = 'icons/mob/inhands/equipment/hydroponics_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/hydroponics_righthand.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	var/paint_color = "#780000"
+	var/static/list/no_draw_turfs = typecacheof(list(/turf/open/space, /turf/open/openspace, /turf/open/lava, /turf/open/chasm))
+
+	/// If you've started drawing a rune you can't start again somewhere else
+	var/obj/effect/decal/cleanable/traitor_rune/rune
+	/// Are we currently drawing? Used to prevent spam clicks for do_while
+	var/drawing_rune = FALSE
+	/// Set to true if we finished drawing something, this spraycan is now useless
+	var/expended = FALSE
+
+/obj/item/traitor_spraycan/afterattack(atom/target, mob/user, proximity, params)
+	. = ..()
+
+	if (expended)
+		user.balloon_alert(user, "all out of paint...")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (drawing_rune)
+		user.balloon_alert(user, "already busy!")
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if (!proximity || !check_allowed_items(target) || !isliving(user))
+		return
+
+	if (isturf(target))
+		try_draw_new_rune(user, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+	else if (target == rune)
+		try_complete_rune(user, target)
+		return COMPONENT_CANCEL_ATTACK_CHAIN
+
+/**
+ * Attempt to draw a rune on [target_turf].
+ * Shamelessly adapted from the heretic rune drawing process.
+ *
+ * Arguments
+ * * user - the mob drawing the rune
+ * * target_turf - the place the rune's being drawn
+ */
+/obj/item/traitor_spraycan/proc/try_draw_new_rune(mob/living/user, turf/target_turf)
+	if (rune)
+		user.balloon_alert(user, "you've already started drawing somewhere else!")
+		return
+
+	var/list/target_turfs = RANGE_TURFS(1, target_turf)
+	for(var/turf/nearby_turf as anything in target_turfs)
+		if (!isopenturf(nearby_turf) || is_type_in_typecache(nearby_turf, no_draw_turfs))
+			user.balloon_alert(user, "you need a clear 3x3 area!")
+			return
+
+	draw_rune(user, target_turf, target_turfs)
+
+#define RUNE_DRAW_STEP 3 SECONDS
+
+/**
+ * Draw your stage one rune on the ground and store it.
+ *
+ * Arguments
+ * * user - the mob drawing the rune
+ * * target_turf - the place the rune's being drawn
+ */
+/obj/item/traitor_spraycan/proc/draw_rune(mob/living/user, turf/target_turf)
+	user.balloon_alert(user, "drawing outline...")
+	drawing_rune = TRUE
+	if ((try_draw_step("drawing outline...", user, target_turf)))
+		rune = new /obj/effect/decal/cleanable/traitor_rune(target_turf)
+		rune.owner = user
+		RegisterSignal(rune, COMSIG_PARENT_QDELETING, .proc/on_rune_destroyed)
+		try_complete_rune(user, rune)
+
+/**
+ * Called when the rune we were drawing is cleaned up.
+ * If you haven't reached the final stage yet this is fine and you can start over.
+ */
+/obj/item/traitor_spraycan/proc/on_rune_destroyed()
+	SIGNAL_HANDLER
+	UnregisterSignal(rune, COMSIG_PARENT_QDELETING)
+	rune = null
```

Comment:
Originally I was doing this so that you didn't draw a bunch of "stage one" outlines everywhere but after thinking about it I guess I don't care if someone decides to do that, it's only likely to harm them in the long run and it still has a 3 second draw time so it's not like they can "spam" them.